src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -100,18 +100,28 @@ Optional<ExternalObjectDirectoryEntity> findMatchingExternalObjectDirectoryEntit
             WHERE eod.status in :failedStatuses
             AND eod.externalLocationType = :type
             AND eod.transferAttempts <= :transferAttempts
+            AND eod.osrUuid is null
             """
     )
     List<ExternalObjectDirectoryEntity> findNotFinishedAndNotExceededRetryInStorageLocation(List<ObjectRecordStatusEntity> failedStatuses,
                                                                                             ExternalLocationTypeEntity type,
                                                                                             Integer transferAttempts,
                                                                                             Pageable pageable);
 
-    default List<ExternalObjectDirectoryEntity> findNotFinishedAndNotExceededRetryInStorageLocation(List<ObjectRecordStatusEntity> failedStatuses,
-                                                                                                    ExternalLocationTypeEntity type,
-                                                                                                    Integer transferAttempts) {
-        return findNotFinishedAndNotExceededRetryInStorageLocation(failedStatuses, type, transferAttempts, Pageable.unpaged());
-    }
+    @Query(
+        """
+            SELECT eod FROM ExternalObjectDirectoryEntity eod
+            WHERE eod.status in :failedStatuses
+            AND eod.externalLocationType = :type
+            AND eod.transferAttempts <= :transferAttempts
+            AND eod.osrUuid is not null
+            """
+    )
+    List<ExternalObjectDirectoryEntity> findNotFinishedAndNotExceededRetryInStorageLocationForDets(List<ObjectRecordStatusEntity> failedStatuses,
```

Comment:
is this different to line 103?

Reply:
yes the difference is whether osrUuid is not null whereas the original one checks if it is null. This maybe the only way to distinguish between migration ARM and DETS ARM